Add `#include <algorithm>` to `jit_compiler.cpp` for Linux builds

Can we add #include <algorithm> to [`/backend/cpu/jit_compiler.cpp`](https://github.com/ml-explore/mlx/blob/main/mlx/backend/cpu/jit_compiler.cpp#L121) because the file uses std::find_if on line 121 but wasn't including the header that defines it. https://github.com/ml-explore/mlx/blob/aaf78f4c6b618daa0fd74697a47cc467920e958f/mlx/backend/cpu/jit_compiler.cpp#L121

The specific error was:
```
mlx/backend/cpu/jit_compiler.cpp:121:12: error: no member named 'find_if' in namespace 'std'
```

Looking at the code, `std::find_if` is used in the compilation process. While this might work on some platforms where <algorithm> is transitively included through other headers, it's not guaranteed by the C++ standard. On Linux with GCC/Clang, the transitive inclusion doesn't happen, causing the build to fail. (it'd be great

--- a/mlx/backend/cpu/jit_compiler.cpp
+++ b/mlx/backend/cpu/jit_compiler.cpp
@@ -2,6 +2,7 @@
 
 #include "mlx/backend/cpu/jit_compiler.h"
 
+#include <algorithm>
 #include <sstream>
 #include <vector>
 

